### PR TITLE
Simple thread pool class.

### PR DESCRIPTION
**Pull request by bwbeach**
I didn't find an obvious library to use, so I wrote a super-simple thread pool.

### CODE REVIEW

File: b2/thread_pool.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+######################################################################
+
+from __future__ import print_function
+
+import threading
+from six.moves import range
+from six.moves.queue import Queue
+
+# this is added to the task queue to tell the threads to stop
+SHUT_DOWN_TOKEN = 'shut_down'
```
Comment (ppolewicz) [inline b2/thread_pool.py]: I suggest to change it to None or an object - comparing strings via `is` operator is not recommended

File: b2/thread_pool.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        """
+        self.queue.put(task)
+
+    def join_all(self):
+        """
+        Waits until all tasks are completed and then shuts down the threads.
+        """
+        # Tell each of the threads to stop.  Each thread will consume
+        # one of the tokens and then exit.
+        for i in range(len(self.threads)):
```
Comment (ppolewicz) [inline b2/thread_pool.py]: Please add a comment here explaining why it is not `for t in self.threads`

File: b2/thread_pool.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            self.queue.put(SHUT_DOWN_TOKEN)
+        self.queue = None
```
Comment (ppolewicz) [inline b2/thread_pool.py]: Why?

File: b2/thread_pool.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+        # Wait for the threads to be done.
+        for thread in self.threads:
+            thread.join()
+        self.threads = None
```
Comment (ppolewicz) [inline b2/thread_pool.py]: Why?

File: b2/thread_pool.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        pool.join_all()
+        errors = pool.get_exceptions()
+    """
+
+    def __init__(self, thread_count=10, queue_capacity=1000):
+        """
+        Initializes a new thread pool, starts the threads running,
+        and gets ready to accept tasks.
+        """
+        self.queue = Queue(maxsize=queue_capacity)
```
Comment (ppolewicz) [inline b2/thread_pool.py]: Those fields are not a part of class interface and should have an underscore prefix

File: b2/thread_pool.py (hunk 1 above)
Comment (bwbeach) [inline b2/thread_pool.py]: Will do.

File: b2/thread_pool.py (hunk 2 above)
Comment (bwbeach) [inline b2/thread_pool.py]: `for _ in self.threads`

File: b2/thread_pool.py (hunk 3 above)
Comment (bwbeach) [inline b2/thread_pool.py]: So that nobody adds any tasks after calling `join_all`.  Will comment.

File: b2/thread_pool.py (hunk 4 above)
Comment (bwbeach) [inline b2/thread_pool.py]: Because there aren't any threads any more.  ThreadPool is intended to a use-once class.

File: b2/thread_pool.py (hunk 5 above)
Comment (bwbeach) [inline b2/thread_pool.py]: yes.

**Pull request merged**